src/routes/movies.py

```
@@ -1,13 +1,330 @@
+
+            if not language:
+                language = LanguageModel(name=language_name)
+                db.add(language)
+                await db.flush()
+
+            languages.append(language)
+
+        # Create movie
+        movie = MovieModel(
+            name=movie_data.name,
+            date=movie_date,
+            score=movie_data.score,
+            overview=movie_data.overview,
+            status=MovieStatusEnum(movie_data.status),
+            budget=movie_data.budget,
+            revenue=movie_data.revenue,
+            country_id=country.id,
+            genres=genres,
+            actors=actors,
+            languages=languages
+        )
+
+        db.add(movie)
+        await db.commit()
+        await db.refresh(movie)
+
+        # Load relationships for response
+        await db.refresh(movie, ['country', 'genres', 'actors', 'languages'])
+
+        return MovieDetailSchema(
+            id=movie.id,
+            name=movie.name,
+            date=movie.date,
+            score=movie.score,
+            overview=movie.overview,
+            status=movie.status.value,
+            budget=float(movie.budget),
+            revenue=movie.revenue,
+            country={
+                "id": movie.country.id,
+                "code": movie.country.code,
+                "name": movie.country.name
+            },
+            genres=[
+                {"id": genre.id, "name": genre.name}
+                for genre in movie.genres
+            ],
+            actors=[
+                {"id": actor.id, "name": actor.name}
+                for actor in movie.actors
+            ],
+            languages=[
+                {"id": lang.id, "name": lang.name}
+                for lang in movie.languages
+            ]
+        )
+
+    except HTTPException:
+        raise
+    except Exception:
+        await db.rollback()
+        raise HTTPException(status_code=400, detail="Invalid input data.")
```

Catching broad Exception and returning a generic 400 hides the root cause and makes debugging harder. Be specific where possible (e.g. catch ValueError for conversion errors, IntegrityError for DB uniqueness errors). At minimum, log the exception before returning the generic 400 so unexpected server errors aren't silently converted to 'Invalid input data.'